tests/test_transpile.py

```diff
@@ -809,3 +809,26 @@ def unsupported(level):
         with self.assertRaises(UnsupportedError) as ctx:
             unsupported(ErrorLevel.IMMEDIATE)
         self.assertEqual(str(ctx.exception).count(error), 1)
+
+    def test_isinf(self):
+        self.assertEqual(
+            transpile("SELECT ISINF(x)", read="duckdb", write="bigquery")[0], "SELECT IS_INF(x)"
+        )
+        self.assertEqual(
+            transpile("SELECT IS_INF(x)", read="bigquery", write="duckdb")[0], "SELECT ISINF(x)"
+        )
+
+    def test_isnan(self):
+        self.assertEqual(
+            transpile("SELECT ISNAN(x)", read="duckdb", write="bigquery")[0], "SELECT IS_NAN(x)"
+        )
+        self.assertEqual(
+            transpile("SELECT IS_NAN(x)", read="bigquery", write="duckdb")[0], "SELECT ISNAN(x)"
+        )
+
+        self.assertEqual(
+            transpile("SELECT ISNAN(x)", read="hive", write="bigquery")[0], "SELECT IS_NAN(x)"
+        )
+        self.assertEqual(
+            transpile("SELECT IS_NAN(x)", read="bigquery", write="hive")[0], "SELECT ISNAN(x)"
+        )
```

Comment:
You can simplify these tests by using helpers such as `validate_all`.

For example, I'd write a test like:

```python
self.validate_all(
    "ISINF(x)",
    read={
        "bigquery": "IS_INF(x)",
    },
    write={
        "bigquery": "IS_INF(x)",
        "duckdb": "ISINF(x)",
    },
)
```

This takes care of DuckDB <-> BigQuery and also DuckDB <-> DuckDB. So if you did this you could get rid of the `self.validate_identity` call you wrote in this PR as well. Similarly for `ISNAN(x)` and `IS_NAN(x)`.

Reply:
It looks like the `self.validate_all` helper is a member of `Validator`, which is [namespaced under `dialects`](https://github.com/tobymao/sqlglot/blob/8574d86ecdb0085e1e496fe8a456254cfac29a19/tests/dialects/test_dialect.py#L27). Should `class TestTranspile` inherit from `dialects.Validator` here, to use the helper? odd to be pulling a dep from a nested package here

Reply:
Nope, the action item here is to get rid of all new tests under `TestTranspile` and add a couple tests in `test_duckdb.py` that look similar to the one above which uses `self.validate_all`

Reply:
The `test_transpile.py` is for more generic and dialect-agnostic (usually) tests

Reply:
ack - making the changes now